File: Kattis/a-multiplication-game.py
import sys

for line in sys.stdin:
    num = int(line)
    counter = 2
    temp = 1
    while temp < num:
        if counter == 2:
            counter = 9
        else:
            counter = 2
        temp *= counter
    if counter == 2:
        print("Ollie wins.")
    else:
        print("Stan wins.")

File: Kattis/C/a_multiplication_game.c
#include <stdio.h>

int main(void) {
    // unsigned long long is used instead of integer to avoid overflow
    // overflow occurs when the number is too large to be stored in the variable
    // long long is at least 64 bits, which is enough to store 2^64 - 1
    unsigned long long n;
    // llu is used to print unsigned long long
    while (scanf("%llu", &n) != EOF) {
        unsigned long long temp = 1;
        while (temp < n) {
            temp *= 9;
            if (temp >= n) {
                printf("Stan wins.\n");
                break;
            }
            temp *= 2;
            if (temp >= n) {
                printf("Ollie wins.\n");
                break;
            }
        }
    }
    return 0;
}